proof/invariant-abstract/ADT_AI.thy

```diff
@@ -11,17 +11,13 @@ imports
   ArchADT_AI
 begin
 
-context begin interpretation Arch .
-
-requalify_consts
+arch_requalify_consts (A)
   empty_context
   init_A_st
+
+arch_requalify_consts
   ptable_lift
```

Comment:
Hmm, I hadn't considered that the new approach of not doing the interpretation means that when requalifying we need to be more specific about the naming. I guess this would fix itself if we were consistent, which the new `arch_global_naming` helps with. Either way, it's still a trade-off that seems worth making.

Reply:
It is a bit annoying. I thought about ways to work around it, but then it turns out that I like the aesthetic of knowing whether something came from ASpec/ExecSpec or the proofs. With a bunch of work we could make the command be able to take the `(A/H)` on a per-id level, e.g. `arch_requalify_consts empty_context (A) ptable_lift init_A_st (A)` or something, but I'm not sure about how beneficial that would be.

Reply:
Ah right, I hadn't stopped to realise that the `A/H` is only on things from the specs. That's not as bad as I first thought then, I was thinking it was due to us not being consistent with naming.

Reply:
Still a bit annoying but I agree, probably not worth making the command more complicated to handle it.